jupyter_client/jsonutil.py

```diff
@@ -75,7 +83,8 @@ def squash_dates(obj):
 def date_default(obj):
     """default function for packing datetime objects in JSON."""
     if isinstance(obj, datetime):
-        return obj.isoformat()
+        obj = _ensure_tzinfo(obj)
```

Comment:
That is the intention. I'm ever so slightly concerned about breaking things with a hard change. How about a warning that this is what's happening? I added a DeprecationWarning, but it could also be one that's on by default.